packages/manager/modules/overthebox/src/overthebox/autoconfigure/dhcp/autoconfigure-dhcp.controller.js

```diff
@@ -0,0 +1,256 @@
+        leaseDuration: this.config.leaseDuration,
+        offset: this.config.offset,
+        poolSize: this.config.poolSize,
+      },
+      priority: this.config.priority,
+    };
+
+    if (this.config.interface) {
+      params.configuration.interface = this.config.interface;
+    }
+
+    return this.$http
+      .post(`/overTheBox/${this.serviceName}/configuration/dhcp/config`, params)
+      .then((response) => {
+        this.successMessage = this.$translate.instant(
+          'overTheBox_autoconfigure_dhcp_config_create_succeed',
+        );
+
+        // Reload config table
+        this.loadConfig();
+        this.showAddConfig = false;
+        this.displayAddConfigButton = true;
+        return response;
+      })
+      .catch((error) => {
+        // Display error message
+        this.errorMessage = this.$translate.instant(
+          'overTheBox_autoconfigure_create_failed',
+          {
+            errorMessage: error.data.message,
+          },
+        );
+      });
+  }
+
+  cancelCreateConfig() {
+    this.showAddConfig = false;
+    this.displayAddConfigButton = true;
+  }
+
+  removeConfig(row) {
+    this.errorMessage = '';
+    this.successMessage = '';
+    return this.$http
+      .delete(
+        `/overTheBox/${this.serviceName}/configuration/dhcp/config/${row.id}`,
+      )
+      .then((response) => {
+        // Display success message
+        this.successMessage = this.$translate.instant(
+          'overTheBox_autoconfigure_dhcp_config_remove_succeed',
+        );
+
+        // Reload config table
+        this.loadConfig();
+        return response;
+      })
+      .catch((error) => {
+        // Display error message
+        this.errorMessage = this.$translate.instant(
+          'overTheBox_autoconfigure_remove_failed',
+          {
+            errorMessage: error.data.message,
+          },
+        );
+      });
+  }
+
+  loadStaticLeases() {
+    this.staticLeases = [];
+    return this.$http
+      .get(`/overTheBox/${this.serviceName}/configuration/dhcp/staticLease`)
+      .then((response) => {
+        if (response.data) {
+          this.staticLeases = response.data;
+        }
+        return response.data;
+      })
+      .catch((error) => {
+        // Display error message
+        this.errorMessage = this.$translate.instant(
+          'overTheBox_autoconfigure_loading_failed',
+          {
+            errorMessage: error.data.message,
+          },
+        );
+      });
+  }
+
+  addStaticLease() {
+    this.staticLease.hostname = '';
+    this.staticLease.ip = '';
+    this.staticLease.mac = '';
+    this.staticLease.priority = '';
+    this.showAddStaticLease = true;
+    this.displayAddStaticLeaseButton = false;
+  }
+
+  createStaticLease() {
+    this.errorMessage = '';
+    this.successMessage = '';
+    if (
+      !this.staticLease.hostname &&
+      !this.staticLease.ip &&
+      !this.staticLease.mac &&
+      !this.staticLease.priority
+    ) {
+      return null;
```

Comment:
```suggestion
      return this.$q.resolve(null);
```
to be consistent with return type if chain is needed

Reply:
There is no chain needed.

Reply:
Even though, it is nice to be consistent in return type